src/psyclone/psyir/transformations/region_trans.py

```diff
@@ -0,0 +1,170 @@
+                                     the transformation acts on the child of \
+                                     a single-line If or Where statment.
+
+        '''
+        # pylint: disable=too-many-branches
+        from psyclone.psyGen import IfBlock, Loop
+        from psyclone.nemo import NemoInvokeSchedule
+        from psyclone.psyir.transformations import TransformationError
+        if not options:
+            options = {}
+        node_parent = node_list[0].parent
+        prev_position = -1
+        for child in node_list:
+            if child.parent is not node_parent:
+                raise TransformationError(
+                    "Error in {0} transformation: supplied nodes "
+                    "are not children of the same parent."
+                    .format(self.name))
+            if prev_position >= 0 and prev_position+1 != child.position:
+                raise TransformationError(
+                    "Children are not consecutive children of one parent: "
+                    "child '{0}' has position {1}, but previous child had "
+                    "position {2}."
+                    .format(str(child), child.position, prev_position))
+            prev_position = child.position
+
+        # Check that the proposed region contains only supported node types
+        if options.get("node-type-check", True):
+            for child in node_list:
+                # Stop at any instance of Kern to avoid going into the
+                # actual kernels, e.g. in Nemo inlined kernels
+                flat_list = [item for item in child.walk(object, Kern)
+                             if not isinstance(item, Schedule)]
+                for item in flat_list:
+                    if not isinstance(item, self.valid_node_types):
+                        raise TransformationError(
+                            "Nodes of type '{0}' cannot be enclosed by a {1} "
+                            "transformation".format(type(item), self.name))
+
+        # If we've been passed a list that contains one or more Schedules
+        # then something is wrong. e.g. two Schedules that are both children
+        # of an IfBlock would imply that the transformation is being applied
+        # around both the if-body and the else-body and that doesn't make
+        # sense.
+        if isinstance(node_list, list) and len(node_list) > 1 and \
+           any([isinstance(node, Schedule) for node in node_list]):
+            raise TransformationError(
+                "Cannot apply a transformation to multiple nodes when one or "
+                "more is a Schedule. Either target a single Schedule or the"
+                " children of a Schedule.")
+
+        # Sanity check that we've not been passed the condition part of
+        # an If statement or the bounds of a Loop. If the parent node is
+        # a Loop of IfBlock then we can only accept a single Schedule.
+        # TODO #542 Once everything has a Schedule we can tidy this up
```

Comment:
#542 has been done and this remains! My fault. Would you mind correcting along the lines of this comment?

Reply:
I hope I did the right thing there.

Reply:
Almost! You just need to tighten-up the second part of the check to make sure that the list has only one member and that that member is a Schedule. (Since e.g. the if-body and else-body nodes are both Schedules but it would be an error to supply both of them to a transformation.)

Reply:
Yes, thank you :-)